ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -120,28 +133,43 @@ func (r *RayClusterReconciler) eventReconcile(request ctrl.Request, event *corev
 	// we only care about pod events
 	if event.InvolvedObject.Kind != "Pod" || event.Type != "Warning" || event.Reason != "Unhealthy" ||
 		!strings.Contains(event.Message, "Readiness probe failed") {
+		// This is not supposed to happen since we already filter events in the watch
+		msg := fmt.Sprintf("unexpected event, we should have already filtered these conditions: %v", event)
+		r.Log.Error(fmt.Errorf(msg), msg, "event", event)
 		return ctrl.Result{}, nil
 	}
 
 	_ = r.Log.WithValues("event", request.NamespacedName)
-	r.Log.Info("reconcile RayCluster Event", "event name", request.Name)
 
-	if err := r.List(context.TODO(), &pods,
-		client.InNamespace(event.InvolvedObject.Namespace)); err != nil {
+	options := []client.ListOption{
+		client.MatchingFields(map[string]string{podUIDIndexField: string(event.InvolvedObject.UID)}),
+		client.InNamespace(event.InvolvedObject.Namespace),
+		client.MatchingLabels(map[string]string{common.RayNodeLabelKey: "yes"}),
```

Comment:
Remove all `common.RayNodeLabelKey`

Reply:
If you have time to help this, that will be good.

Reply:
why?

Reply:
I found out that I misunderstood some things. Initially, I thought that we should implement all filter logic in the resource event handler (i.e. `CreateFunc`) and return true only when an event is issued by a Ray Pod. With this approach, we can make the function `eventReconcile` only handle event processing and not worry about event filtering. Therefore, the label `common.RayNodeLabelKey` is not necessary.

However, I have since realized that the owners of the unhealthy events are Pods, which may include Ray Pods and other Pods. Therefore, we cannot use `GetControllerOf` to get the owner and determine whether the event belongs to a Ray Pod or not by checking whether the owner's kind is "RayCluster" or not.

Hence, the implementation will be more complex. If we want to distinguish whether an event belongs to a Ray Pod or not, we need to

* Get the UID of the event's InvolvedObject.
* Use the UID to look up Pods from the local cache.
* Use `GetControllerOf(pod)` to get the owner, and check whether the `kind` is `RayCluster` or not.

I will make the function `eventReconcile` easier to understand, but we need to have more I/O (read local cache) which may cause performance overhead. 

In summary, we should use `common.RayNodeLabelKey` to distinguish Ray Pods in `eventReconcile` until we understand the cost of the look-up operation.